sdk/typescript/src/zklogin/publickey.ts

```diff
@@ -75,5 +96,84 @@ export function toZkLoginPublicIdentifier(
 	tmp.set([issBytes.length], 0);
 	tmp.set(issBytes, 1);
 	tmp.set(addressSeedBytesBigEndian, 1 + issBytes.length);
-	return new ZkLoginPublicIdentifier(tmp);
+	return new ZkLoginPublicIdentifier(tmp, options);
+}
+
+const VerifyZkLoginSignatureQuery = graphql(`
+	query Zklogin(
+		$bytes: Base64!
+		$signature: Base64!
+		$intent_scope: ZkLoginIntentScope!
+		$author: SuiAddress!
+	) {
+		verifyZkloginSignature(
+			bytes: $bytes
+			signature: $signature
+			intentScope: $intent_scope
+			author: $author
+		) {
+			success
+			errors
+		}
+	}
+`);
+
+async function graphqlVerifyZkLoginSignature({
+	address,
+	bytes,
+	signature,
+	client = new SuiGraphQLClient({
+		url: 'https://sui-mainnet.mystenlabs.com/graphql',
+	}),
+}: {
+	address: string;
+	bytes: string;
+	signature: string;
+	client?: SuiGraphQLClient;
+}) {
+	const resp = await client.query({
+		query: VerifyZkLoginSignatureQuery,
+		variables: {
+			bytes,
+			signature,
+			intent_scope: 'PERSONAL_MESSAGE',
```

Comment:
Could we pass in the intentScope so that we can implement verifyPersonalMessage and verifyTransactionBlock on the PublicKey class?

Reply:
good point! revised and added tests